ProcessLib/RichardsFlow/RichardsFlowFEM.h

```diff
@@ -121,10 +120,11 @@ class LocalAssemblerData : public RichardsFlowLocalAssemblerInterface
                 dSwdPc = interpolated_Pc.getDerivative(
                     interpolated_Pc.getSupportMin());
 
+            double const Sw = interpolated_Pc.getValue(Pc);
             _saturation[ip] = Sw;
 
-            double k_rel = interpolated_Kr.getValue(Sw);
-            double drhow_dp(0.0);
+            double const k_rel = interpolated_Kr.getValue(Sw);
+            double const drhow_dp(0.0);
```

Comment:
I do not see where this is commented in the code, but you are right, the `mass_mat_coeff` is not zero. I am unsure how the compiler handles the expression `poro  * Sw * drhow_dp` and how important the optimization of the calculation of this expression is.